File: music-application/src/main/java/com/music/sale/application/user/dto/input/AgreeTermsInput.java
package com.music.sale.application.user.dto.input;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Builder;

@Builder
public record AgreeTermsInput(
    @NotNull Long userId,
    @NotBlank String title,
    @NotBlank String version,
    @NotNull Boolean isAgreed) {}

File: music-application/src/main/kotlin/com/music/sale/application/user/dto/input/AgreeTermsInput.kt
package com.music.sale.application.user.dto.input

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull

data class AgreeTermsInput(
    @field:NotNull val userId: Long,
    @field:NotBlank val title: String,
    @field:NotBlank val version: String,
    @field:NotNull val isAgreed: Boolean,
)
